app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/PuzzleFragment.kt

```diff
@@ -0,0 +1,285 @@
+import io.zenandroid.onlinego.R
+import io.zenandroid.onlinego.gamelogic.RulesManager
+import io.zenandroid.onlinego.utils.showIf
+import io.zenandroid.onlinego.ui.composables.Board
+import io.zenandroid.onlinego.ui.composables.RatingBar
+import io.zenandroid.onlinego.ui.screens.main.MainActivity
+import io.zenandroid.onlinego.ui.screens.puzzle.PuzzleAction.*
+import io.zenandroid.onlinego.ui.theme.OnlineGoTheme
+import io.zenandroid.onlinego.data.model.StoneType
+import io.zenandroid.onlinego.data.model.local.Puzzle
+import io.zenandroid.onlinego.data.model.local.PuzzleCollection
+import io.zenandroid.onlinego.mvi.MviView
+import io.zenandroid.onlinego.data.repositories.SettingsRepository
+import io.zenandroid.onlinego.utils.PersistenceManager
+import io.zenandroid.onlinego.utils.convertCountryCodeToEmojiFlag
+import org.commonmark.node.*
+import org.koin.android.ext.android.inject
+import org.koin.android.ext.android.get
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import java.time.Instant.now
+import java.time.temporal.ChronoUnit.*
+import org.koin.core.parameter.parametersOf
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.res.painterResource
+import io.zenandroid.onlinego.data.model.BoardTheme
+import io.zenandroid.onlinego.gamelogic.Util.toCoordinateSet
+import io.zenandroid.onlinego.utils.analyticsReportScreen
+import io.zenandroid.onlinego.utils.rememberStateWithLifecycle
+
+const val COLLECTION_ID = "COLLECTION_ID"
+
+private const val TAG = "PuzzleFragment"
+
+@ExperimentalAnimationApi
+@ExperimentalMaterialApi
+@ExperimentalFoundationApi
+@ExperimentalComposeUiApi
+class PuzzleFragment : Fragment(), MviView<PuzzleState, PuzzleAction> {
+    private val settingsRepository: SettingsRepository by inject()
+    private val viewModel: PuzzleViewModel by viewModel {
+        parametersOf(arguments!!.getLong(COLLECTION_ID))
+    }
+
+    private val internalActions = PublishSubject.create<PuzzleAction>()
+    private var currentState: PuzzleState? = null
+
+    private val onBackPressedCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() {
+            internalActions.onNext(UserPressedBack)
+            findNavController().navigateUp()
+        }
+    }
+
+    @Composable
+    private fun Header(text: String) {
+        Text(
+            text = text,
+            fontSize = 12.sp,
+            color = MaterialTheme.colors.onBackground,
+            fontWeight = FontWeight.Medium,
+            modifier = Modifier.padding(start = 8.dp, top = 8.dp, bottom = 8.dp)
+        )
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(
+                DisposeOnLifecycleDestroyed(viewLifecycleOwner)
+            )
+            setContent {
+                OnlineGoTheme {
```

Comment:
in order to ensure that the composable does not mistakenly use any state from the Fragment, I like to bring the Composable out of the fragment scope, sometimes even in a different file (e.g. `PuzzleScreenUI.kt`).